### PR TITLE
fix: ImageHelper - new methods with ImageSize parameters + minor fix

### PR DESCRIPTION
**Pull request by monsieurtanuki**
Impacted file:
* `ImageHelper.dart`: added flexibility and `ImageSize` parametered methods; fixed a minor bug (trailing slash)

### What
- Refactoring around `ImageHelper`

### Part of 
- https://github.com/openfoodfacts/smooth-app/issues/3065
- https://github.com/openfoodfacts/smooth-app/pull/3088

### CODE REVIEW

**Comment by WildOrangutan** [comment]
Would it be possible to shorten function naming a bit. Some are getting overly verbose imho.

I suggest making `imageSize` as optional argument(s) and taking it from `ProductImage` if not set.

**Comment by monsieurtanuki** [comment]
> Would it be possible to shorten function naming a bit. Some are getting overly verbose imho.
> I suggest making `imageSize` as optional argument(s) and taking it from `ProductImage` if not set.

Sounds reasonable.

File: lib/utils/ImageHelper.dart [hunk 1]
```diff
@@ -27,23 +27,31 @@ class ImageHelper {
     }
   }
 
-  /// Returns the product image full url, or null if [barcode] is null
+  /// Returns the [image] full url - for a specific [imageSize] if needed.
```
Comment (M123-dev) [inline lib/utils/ImageHelper.dart:30]:
```suggestion
  /// Returns the image full url as [String] - for a specific [imageSize] if needed.
```

File: lib/utils/ImageHelper.dart [hunk 2]
```diff
@@ -27,23 +27,31 @@ class ImageHelper {
...
   }) =>
       barcode == null
           ? null
-          : '${getProductImageRootUrl(barcode, queryType: queryType)}/${image.field.value}_${image.language.code}.${image.rev}.${image.size.toNumber()}.jpg';
+          : '${getProductImageRootUrl(barcode, queryType: queryType)}'
+              '/'
+              '${getProductImageFilename(image, imageSize: imageSize)}';
 
-  /// Returns the product image filename
+  /// Returns the [image] filename - for a specific [imageSize] if needed.
```
Comment (M123-dev) [inline lib/utils/ImageHelper.dart:46]: Same here

**Review by M123-dev — APPROVED** [review]
LGTM, thanks @monsieurtanuki

**Pull request merged**